src/main/xerus/monstercat/api/Playlist.kt

```diff
@@ -86,4 +101,181 @@ object Playlist {
+				}
+			}
+		}
+		fun delete(runAfter: () -> Unit = {}) {
+			if (connectTable.selectionModel.selectedItem != null) {
+				GlobalScope.async {
+					APIConnection.editPlaylist(connectTable.selectionModel.selectedItem.id, deleted = true)
+					onFx { runAfter.invoke() }
+				}
+			}
+		}
+		fun new(runAfter: () -> Unit = {}) {
+			val subParent = VBox()
+			val subStage = stage.createStage("New Playlist", subParent)
+			subStage.initModality(Modality.WINDOW_MODAL)
+			val textField = TextField().apply { promptText = "Name" }
+			val publicTick = CheckBox("Public")
+			subParent.children.addAll(textField, publicTick)
+			subParent.addRow(createButton("Create"){
+				GlobalScope.async {
+					APIConnection.createPlaylist(textField.text.let { if (it.isBlank()) "New Playlist" else it }, Playlist.tracks, publicTick.isSelected)
+					onFx { subStage.close(); runAfter.invoke() }
+				}
+			}, createButton("Cancel"){
+				subStage.close()
+			})
+			subStage.show()
+		}
+		fun rename(runAfter: () -> Unit = {}) {
+			val subParent = VBox()
+			val subStage = stage.createStage("Rename Playlist", subParent)
+			subStage.initModality(Modality.WINDOW_MODAL)
+			val textField = TextField().apply { promptText = "Name" }
+			subParent.add(textField)
+			subParent.addRow(createButton("Rename"){
+				if (connectTable.selectionModel.selectedItem != null) {
+					GlobalScope.async {
+						APIConnection.editPlaylist(connectTable.selectionModel.selectedItem.id, name = textField.text.let { if (it.isBlank()) "Unnamed" else it })
+						onFx { subStage.close(); runAfter.invoke() }
+					}
+				}
+			}, createButton("Cancel"){
+				subStage.close()
+			})
+			subStage.show()
+		}
+
+		val playlists = FXCollections.observableArrayList<ConnectPlaylist>()
+		fun updatePlaylists() {
+			playlists.clear()
+			if (APIConnection.connectValidity.value != ConnectValidity.NOGOLD && APIConnection.connectValidity.value != ConnectValidity.GOLD){
+				connectTable.placeholder = Label("Please connect using connect.sid in the downloader tab.")
+			}else{
+				connectTable.placeholder = Label("Loading...")
+				GlobalScope.async {
+					val results = connection.getPlaylists()
+					if (results != null && results.isNotEmpty())
+						playlists.addAll(results)
+					else
+						onFx {
+							connectTable.placeholder = Label("No playlists were found on your account.")
+						}
+				}
+			}
+		}
+
+		connectTable.apply {
+
+			columnResizePolicy = TableView.CONSTRAINED_RESIZE_POLICY
+			columns.addAll(TableColumn<ConnectPlaylist, String>("Name").apply {
+				cellValueFactory = Callback<TableColumn.CellDataFeatures<ConnectPlaylist, String>, ObservableValue<String>> { p ->
+					ImmutableObservable(p.value.name)
+				}
+			}, TableColumn<ConnectPlaylist, String>("Size").apply {
+				cellValueFactory = Callback<TableColumn.CellDataFeatures<ConnectPlaylist, String>, ObservableValue<String>> { p ->
+					ImmutableObservable(p.value.tracks.size.toString())
+				}
+			})
+
+			items = playlists
```

Comment:
I'd prefer if you could logically group the initialisation code

Reply:
587ecea842105969f76a66c0ff4a215b4eca12f4